File: src/main/java/com/lcaohoanq/nocket/domain/cache/IPostRedisService.java
package com.lcaohoanq.nocket.domain.cache;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.lcaohoanq.nocket.domain.post.PostPort.PostResponse;
import java.util.List;
import org.springframework.data.domain.PageRequest;

public interface IPostRedisService {
    
    void clear(); //Clear cached data in Redis
    List<PostResponse> getAllPosts(PageRequest pageRequest) throws JsonProcessingException;
    void saveAllPosts(List<PostResponse> postResponses, PageRequest pageRequest) throws JsonProcessingException;
}

File: src/main/kotlin/com/lcaohoanq/nocket/domain/cache/IPostRedisService.kt
package com.lcaohoanq.nocket.domain.cache

import com.fasterxml.jackson.core.JsonProcessingException
import com.lcaohoanq.nocket.domain.post.PostPort.PostResponse
import org.springframework.data.domain.PageRequest

interface IPostRedisService {
    fun clear() //Clear cached data in Redis

    @Throws(JsonProcessingException::class)
    fun getAllPosts(pageRequest: PageRequest): List<PostResponse>?

    @Throws(JsonProcessingException::class)
    fun saveAllPosts(postResponses: List<PostResponse>, pageRequest: PageRequest)
}
